src/tpm/android/knox/java/CryptoManager.java

```diff
@@ -0,0 +1,423 @@
+        byteBuffer.put(iv);
+        byteBuffer.put(encryptedData);
+        return byteBuffer.array();
+    }
+
+    /**
+     * Decrypts the given encrypted data using a symmetric key stored in the Android KeyStore.
+     * <p>
+     * This method takes encrypted data as input and decrypts it using a symmetric key retrieved from the Android KeyStore.
+     * The decryption process supports GCM, CBC and CTR transformations. The initialization vector (IV)
+     * is extracted from the beginning of the encrypted data. The method initializes a {@link Cipher} instance with the appropriate
+     * transformation, loads the Android KeyStore, retrieves the symmetric key, and initializes the cipher in decryption mode with the
+     * retrieved key and the extracted IV. Finally, the encrypted data is decrypted using the cipher's {@code doFinal} method, and the
+     * original plaintext data is returned as a byte array.
+     *
+     * @param encryptedData The encrypted data to be decrypted, represented as a byte array.
+     * @return A byte array representing the decrypted data.
+     * @throws NoSuchPaddingException             if the requested padding scheme is not available.
+     * @throws NoSuchAlgorithmException           if the requested algorithm is not available.
+     * @throws CertificateException               if there is an issue loading the certificate chain.
+     * @throws IOException                        if there is an I/O error during the operation.
+     * @throws InvalidAlgorithmParameterException if the IV parameter is invalid.
+     * @throws InvalidKeyException                if the key cannot be cast to a SecretKey.
+     * @throws UnrecoverableKeyException          if the key cannot be recovered from the keystore.
+     * @throws KeyStoreException                  if there is an error accessing the keystore.
+     * @throws IllegalBlockSizeException          if the data length is invalid for the decryption algorithm.
+     * @throws BadPaddingException                if the data could not be padded correctly for decryption.
+     * @throws InvalidKeySpecException            if the key specification is invalid.
+     * @throws NoSuchProviderException            if the requested security provider is not available.
+     */
+    public byte[] decryptData(byte[] encryptedData) throws NoSuchPaddingException, NoSuchAlgorithmException,
+            CertificateException, IOException, InvalidAlgorithmParameterException, InvalidKeyException,
+            UnrecoverableKeyException, KeyStoreException, IllegalBlockSizeException, BadPaddingException,
+            InvalidKeySpecException, NoSuchProviderException {
+        keyStore.load(null);
+        SecretKey secretKey = (SecretKey) keyStore.getKey(KEY_NAME, null);
+        String TRANSFORMATION = buildTransformation(secretKey);
+        Cipher cipher = Cipher.getInstance(TRANSFORMATION);
+        ByteBuffer byteBuffer = ByteBuffer.wrap(encryptedData);
```

Comment:
Add a check to ensure that the file is at least IV size in length. Also, I see magic numbers such as 12 being used throughout the code. Please refactor those magic numbers into constants and reference the constants.